apps/api/src/user/service/user.service.ts

```diff
@@ -60,6 +127,95 @@ export class UserService {
     })
   }
 
+  async validateEmailChangeOtp(user: User, otp: string): Promise<User> {
+    const userEmailChange = await this.prisma.userEmailChange.findUnique({
```

Comment:
This find query will now progress like this:
- Try finding the otp+userId combo.
- If found, check if the otp is still valid
- If yes, search for the entry in UserEmailChange table that has otpId = otp.id
- Make the changes

Reply:
Done